src/Illuminate/Validation/Validator.php

```diff
@@ -747,7 +747,7 @@ protected function validateDifferent($attribute, $value, $parameters)
 
         $other = Arr::get($this->data, $parameters[0]);
 
-        return isset($other) && $value != $other;
+        return isset($other) && $value !== $other;
```

Comment:
Imagine we have a form with a couple of `<input type="text">`, for names e.g.
We type "100" in the first one and "1e2" in the second one. 
Why should `different` rule fail in this case?

Could you provide an example where strict check is wrong?